h/api/test/uri_test.py

```diff
@@ -154,6 +153,13 @@
     ("http://example.com?gclid_foo=abcde", "http://example.com?gclid_foo=abcde"),
     ("http://example.com?bar_gclid=abcde", "http://example.com?bar_gclid=abcde"),
     ("http://example.com?WT=abcde", "http://example.com?WT=abcde"),
-])
+]
+
+
+@pytest.mark.parametrize("url_in,url_out", TEST_URLS)
 def test_normalize(url_in, url_out):
     assert uri.normalize(url_in) == url_out
+
+@pytest.mark.parametrize("url,_", TEST_URLS)
```

Comment:
What's the reasoning for not testing on all the URIs?
I just think that the ~240ms (just did a quick test, three cases vs. all the urls) speed penalty is justified because it makes it harder to introduce a bug which wouldn't return a unicode string.